modules/fru.c

```diff
@@ -295,6 +303,11 @@ IPMI_HANDLER(ipmi_storage_read_fru_data_cmd, NETFN_STORAGE, IPMI_READ_FRU_DATA_C
     offset = (req->data[2] << 8) | (req->data[1]);
 
     count = fru_read( fru_id, &(rsp->data[len+1]), offset, count );
+    if (count == 0) {
```

Comment:
Add a comment here explaining that if count == 0, then it may indicate that the fru_read function failed due to a timeout or some I2C error.